aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -352,7 +352,7 @@ Tensor& empty_out(IntArrayRef size,
     return self.to(ScalarType::n, non_blocking);                 \
   }
 
-AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, DEFINE_CAST_OP)
+AT_FORALL_SCALAR_TYPES_AND(Half, DEFINE_CAST_OP)
```

Comment:
Err? What? `Bool` is not a part of `AT_FORALL_SCALAR_TYPES` nor is the BFloat16, so why are you removing those?

Reply:
The generated functions for Bool and BFloat16 are not declared in corresponding yamls, I suspected that they are not being used anywhere.

Reply:
I guarded the definition around clang diagnostic ignored